core/domain/state_domain_test.py

```diff
@@ -212,6 +213,24 @@ def test_state_operations(self):
         exploration.delete_state('END')
         self.assertNotIn('END', exploration.states)
 
+    def test_update_solicit_answer_details(self):
+        """Test updating solicit_answer_details."""
+        state = state_domain.State.create_default_state('state_1')
+        self.assertEqual(
+            state.solicit_answer_details, False)
+        state.update_solicit_answer_details(True)
+        self.assertEqual(
+            state.solicit_answer_details, True)
+
+    def test_update_solicit_answer_details_with_non_bool_fails(self):
+        """Test updating solicit_answer_details with non bool value."""
+        exploration = exp_domain.Exploration.create_default_exploration('eid')
+        init_state = exploration.states[exploration.init_state_name]
```

Comment:
Check the initial value and make sure it's not changed at the end.

Reply:
Done